src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototype.cs

```diff
@@ -1772,6 +1772,11 @@ public DatabasePrototype(CDataContainer context)
                     // Azure instances don't have a model database we can query, so just use the defaults
                     this.originalState = new DatabaseData(context);
                 }
+                else if (context.Server.DatabaseEngineEdition == DatabaseEngineEdition.SqlManagedInstance)
+                {
+                    // SQl Managed Instance don't have a model database we can query, so just use the defaults to master
+                    this.originalState = new DatabaseData(context, "master");
```

Comment:
Is this what SSMS does?

@erinstellato-ms Does this seem like a reasonable approach? I can't remember if MI users will always be guaranteed to have access to master - do we want to consider either using the plain defaults like Azure does?

Reply:
I'm not sure how they handled in SSMS, but only approach that works in our case is by using the master database. I can check if we can use the default like azure.

Reply:
@Charles-Gagnon @ssreerama  I'm checking, give me a few

Reply:
Also do we need to specify master, or can we just do the defaults like the Azure if-case directly above this?

Reply:
It makes sense to try and use "reasonable" defaults when we can, but my question is whether master is actually going to represent reasonable defaults. If it isn't then yeah, I think your suggestion Cory is what I'd lean towards.

Reply:
It worked with or without specifying master as default. I'll check with Erin to confirm if we have any MI specific defaults or this approach is the correct one. Merging this one for now. Thank you all